### PR TITLE
Fix NPE with async Jersey resource

### LINKED ISSUE #2910: CompletionStage results in NPE
[//]: # "Please remove these comments"
[//]: # "Provide a general summary of the issue in the Title above"

## Environment Details
* Helidon Version: 2.2.2
* Helidon SE or Helidon MP: MP
* JDK version: 11
* OS: Windows
* Docker version (if applicable): N/A

----------

## Problem Description
If you use a `CompletionStage` as a response it results in a NPE:

```
Apr 04, 2021 7:09:57 PM io.helidon.webserver.RequestRouting$RoutedRequest defaultHandler
WARNING: Default error handler: Unhandled exception encountered.
java.util.concurrent.ExecutionException: Unhandled 'cause' of this exception encountered.
        at io.helidon.webserver.RequestRouting$RoutedRequest.defaultHandler(RequestRouting.java:397)
        at io.helidon.webserver.RequestRouting$RoutedRequest.nextNoCheck(RequestRouting.java:377)
        at io.helidon.webserver.RequestRouting$RoutedRequest.next(RequestRouting.java:420)
        at io.helidon.webserver.jersey.ResponseWriter.failure(ResponseWriter.java:172)
        at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:438)
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:263)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234)
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680)
        at io.helidon.webserver.jersey.JerseySupport$JerseyHandler.lambda$doAccept$3(JerseySupport.java:293)
        at io.helidon.common.context.Contexts.runInContext(Contexts.java:117)
        at io.helidon.common.context.ContextAwareExecutorImpl.lambda$wrap$5(ContextAwareExecutorImpl.java:154)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NullPointerException
        at org.glassfish.jersey.internal.PropertiesResolver$1.resolveProperty(PropertiesResolver.java:88)
        at org.glassfish.jersey.internal.PropertiesResolver$1.resolveProperty(PropertiesResolver.java:83)
        at org.glassfish.jersey.server.ContainerRequest.resolveProperty(ContainerRequest.java:292)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.unwrapInvocableResponseType(ResourceMethodInvoker.java:489)  
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.lambda$invoke$2(ResourceMethodInvoker.java:468)
        at org.glassfish.jersey.process.internal.Stages$LinkedStage.apply(Stages.java:284)
        at org.glassfish.jersey.process.internal.Stages.process(Stages.java:147)
        at org.glassfish.jersey.server.ServerRuntime$Responder.processResponse(ServerRuntime.java:371)
        at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:363)
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:258)
        ... 14 more
```

## Steps to reproduce
1. Use `helidon-mp-quickstart`
2. Create a resource that responds with `CompletionStage`:
```
	@GET
	@Path("foo")
	@Produces(MediaType.TEXT_PLAIN)
	public CompletionStage<String> foo() {
		return CompletableFuture.completedStage("bar");
	}
```

### PR DESCRIPTION
**Pull request by danielkec**
Fixes #2910

Signed-off-by: Daniel Kec <daniel.kec@oracle.com>

### CODE REVIEW

File: microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class AsyncResourceTest extends Application {
```
Comment (spericas) [inline microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java:44]: Why extend `Application` here?

Comment (tomas-langer) [inline microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java:44]: This class should be using `@HelidonTest`, which allows you to get `WebTarget` with the correct endpoint either in constructor of the test class, or as a parameter of your test method.

File: microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            assertThat(getResponse, is(equalTo("testResponseGet")));
+
+            String putResponse = target.path("/async-app/asyncPut").request().put(Entity.text(""), String.class);
+            assertThat(putResponse, is(equalTo("testResponsePut")));
+        } finally {
+            server.stop();
+        }
+    }
+
+    private final class TestApp extends Application {
```
Comment (tomas-langer) [inline microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java:78]: This is not needed at all, you just need the test resource.

File: microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+        @Override
+        public Set<Object> getSingletons() {
+            return Set.of(new TestResource());
+        }
+    }
+
+    @Path("/")
+    public static final class TestResource {
```
Comment (tomas-langer) [inline microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java:87]: You can add this class using `@AddBean` when using `@HelidonTest`

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java (hunk 1 above)
Comment (danielkec) [reply microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java]: nice thx!

Comment (danielkec) [reply microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java]: yay leftover thx

File: microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java (hunk 2 above)
Comment (danielkec) [reply microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java]: fixed

File: microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java (hunk 3 above)
Comment (danielkec) [reply microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java]: fixed

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**